### PR TITLE
refactoring; cvh integration

### PR DESCRIPTION
**Pull request by gabrielsaruhashi**

### CODE REVIEW

File: src/pages/Call/index.tsx [hunk 1]
```diff
@@ -65,9 +49,16 @@ const CallBase: React.FC<PropsFromRedux & RouteComponentProps<TParams>> = ({
   }, [stableDispatch]);
 
   useEffect(() => {
-    if (!call || hasInit) return;
-    stableDispatch(initializeVisit({ callId: call.id, setRc }));
-  }, [call, stableDispatch, hasInit]);
+    if (!call || hasInit || !call.videoHandler) return;
+    stableDispatch(
```
Comment (gabrielsaruhashi) [inline src/pages/Call/index.tsx:53]: @mpekala23 how have u gone around the dispatch rerendering problem?
rn ti seems like our dispatch is unstable; it keeps rerendering

File: src/api/User.ts [hunk 2]
```diff
@@ -46,5 +46,9 @@ export async function loginWithCredentials(cred: {
     }),
   });
   const body = await response.json();
-  await initializeSession(body.data, cred.language);
+  const cookies = response.headers.get("cookie") || "";
+  const re = /(?<=connect.sid=)([^\s;]+)/gm;
```
Comment (lspdrz) [inline src/api/User.ts:50]: Nice little regex solution for now

**Review by lspdrz — APPROVED** [review]

File: src/components/Dashboard/CallItem.tsx [hunk 3]
```diff
@@ -57,6 +57,7 @@ const CallItem: React.FC<Props> = ({
               size="large"
               type="primary"
               className="rounded-sm"
+              disabled={!!!call.videoHandler}
```
Comment (mpekala23) [inline src/components/Dashboard/CallItem.tsx:60]: `!call.videoHandler`?

File: src/components/Dashboard/ConnectionItem.tsx [hunk 4]
```diff
@@ -13,10 +13,12 @@ const ConnectionItem: React.FC<Props> = ({ contact }) => {
...
-    return differenceInDays(
-      new Date(),
-      new Date(contact.lastCall.scheduledEnd)
-    );
+    // TODO: get the last call
+    // return differenceInDays(
+    //   new Date(),
+    //   new Date(contact.lastCall.scheduledEnd)
+    // );
+    return 1;
```
Comment (mpekala23) [inline src/components/Dashboard/ConnectionItem.tsx:21]: is this something we can comment back in now?

File: src/hooks/useCalls.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  selectAllCalls,
+  selectCallById,
+  selectContactEntities,
+} from "src/redux/selectors";
+import { useAppSelector } from "src/redux";
+
+export function useCalls() {
+  const [calls, setCalls] = useState<Call[]>([]);
+  const baseCalls = useAppSelector(selectAllCalls);
+  const contactEntities = useAppSelector(selectContactEntities);
```
Comment (mpekala23) [inline src/hooks/useCalls.ts:16]: just out of curiousity, how is the store structured? what is a base call, compared to the calls that will be returned from this hook?

File: src/pages/Call/index.tsx (hunk 1 above)
Comment (mpekala23) [reply src/pages/Call/index.tsx:53]: i believe I just don't include dispatch in the dependencies of the `useEffect`?

File: src/pages/Call/index.tsx [hunk 6]
```diff
@@ -65,9 +49,16 @@ const CallBase: React.FC<PropsFromRedux & RouteComponentProps<TParams>> = ({
...
+      initializeVisit({
+        callId: call.id,
+        setRc,
+        authInfo,
+        videoHandler: call.videoHandler,
+      })
+    );
+  }, [call, stableDispatch, hasInit, authInfo]);
```
Comment (mpekala23) [inline src/pages/Call/index.tsx:61]: i think in mobile it would just be `[call, hasInit, authInfo]`

File: src/pages/Calls/index.tsx [hunk 7]
```diff
@@ -1,21 +1,14 @@
...
-import { selectAllCalls, selectUpcomingCalls } from "src/redux/selectors";
+import { useAppSelector } from "src/redux";
 import "src/i18n/config";
 import Calls from "src/components/Calls";
+import { useCalls } from "src/hooks/useCalls";
 
-const mapStateToProps = (state: RootState) => ({
-  user: state.session.user,
-  calls: selectAllCalls(state),
-});
```
Comment (mpekala23) [inline src/pages/Calls/index.tsx:11]: yay!!!

File: src/pages/Dashboard/index.tsx [hunk 8]
```diff
@@ -1,40 +1,34 @@
...
-  calls: selectAllCalls(state),
-  connections: selectAllContacts(state),
-  firstName: state.session.user.firstName,
-});
+import { useCalls } from "src/hooks/useCalls";
 
 const mapDispatchToProps = { fetchCalls, push };
 
-const connector = connect(mapStateToProps, mapDispatchToProps);
+const connector = connect(null, mapDispatchToProps);
```
Comment (mpekala23) [inline src/pages/Dashboard/index.tsx:15]: replace the connector with `useAppDispatch` or `useAppSelector`?

File: src/redux/modules/call.ts [hunk 9]
```diff
@@ -3,59 +3,57 @@ import {
...
 import io from "socket.io-client";
 import { showToast } from "src/utils";
+import { AuthInfo } from "src/types/Session";
 
 export const fetchCalls = createAsyncThunk("calls/fetchAll", async () => {
   const body = await fetchAuthenticated(`calls`);
 
-  const calls = ((body.data as Record<string, unknown>)
-    .calls as BaseCall[]).map((call) => camelcaseKeys(call));
+  const calls = (body.data as Record<string, unknown>).results as BaseCall[];
```
Comment (mpekala23) [inline src/redux/modules/call.ts:22]: `const calls = (body.data as {results: BaseCall[]}).results;`

Comment (mpekala23) [reply src/redux/modules/call.ts:22]: just a little shorter and more clear imo

File: src/types/Call.ts [hunk 10]
```diff
@@ -12,13 +18,15 @@ export interface BaseCall {
...
   connectionId: number;
-  userParticipants: User[];
+  userIds: number[];
   kioskId: number;
+  videoHandler?: CallHandler;
 }
 
 export interface Call extends BaseCall {
-  // connection: Connection;
+  // TODO: load facility kiosks and load kiosk information here
```
Comment (mpekala23) [inline src/types/Call.ts:27]: GH issue?

File: src/types/User.ts [hunk 11]
```diff
@@ -20,9 +20,5 @@ export interface Contact extends BasePersona {
   status: string;
   statusDetails: string;
   relationship: string;
-  lastCall: {
-    id: number;
-    scheduledStart: Date;
-    scheduledEnd: Date;
-  };
+  lastCallId?: number;
```
Comment (mpekala23) [inline src/types/User.ts:23]: this is just to save the work of having to computer this anytime we want to display the last call info?

File: src/utils/Call.ts [hunk 12]
```diff
@@ -1,11 +1,41 @@
...
+import { Dictionary } from "@reduxjs/toolkit";
+import { BaseCall, Call, MediaType } from "src/types/Call";
+import { Contact } from "src/types/User";
+import { notEmpty } from "./utils";
 
 // The timer will switch to a white background for four seconds around each of these key minutes
 const keyMinutes = [10, 5, 2];
 const DEFAULT_BACKGROUND = "bg-gray-900 text-white";
 const WHITE_BACKGROUND = "bg-white text-black";
 const BLUE_BACKGROUND = "bg-blue-500 text-white";
```
Comment (mpekala23) [inline src/utils/Call.ts:10]: heck yeah

**Review by mpekala23 — APPROVED** [review]
a few questions / suggestions, but nothing that should be bloking

File: src/pages/Call/index.tsx (hunk 6 above)
Comment (gabrielsaruhashi) [reply src/pages/Call/index.tsx:61]: doesnt it complain about dispatch @mpekala23 ? 
it's so annoying

File: src/pages/Dashboard/index.tsx (hunk 8 above)
Comment (gabrielsaruhashi) [reply src/pages/Dashboard/index.tsx:15]: will do later if that's okay! 

Need to refactor the structure of the DOC dashboard to match that of Pathways

File: src/redux/modules/call.ts (hunk 9 above)
Comment (gabrielsaruhashi) [reply src/redux/modules/call.ts]: oh i like that

File: src/hooks/useCalls.ts (hunk 5 above)
Comment (gabrielsaruhashi) [reply src/hooks/useCalls.ts:16]: image

File: src/components/Dashboard/ConnectionItem.tsx (hunk 4 above)
Comment (gabrielsaruhashi) [reply src/components/Dashboard/ConnectionItem.tsx:21]: naw @lspdrz has an open PR

File: src/types/User.ts (hunk 11 above)
Comment (gabrielsaruhashi) [reply src/types/User.ts:23]: ye! that's the idea i believe
@lspdrz added it, he said it's better if the server computes it

File: src/utils/Call.ts (hunk 12 above)
Comment (gabrielsaruhashi) [reply src/utils/Call.ts:10]: copy pasta

**Review by gabrielsaruhashi — COMMENTED** [review]
comments to ur qs! @mpekala23

**Pull request merged**